File: Php/userService.php
<?php
// class dengan nama userService
class userService
{
    // variabel2 yang akan digunakan
    protected $email;    // using protected so they can be accessed
    protected $password; // and overidden if necessary
    protected $dataUser; // dummy data
    protected $getRole;  // stores the role data

    // membuat constructor baru dengan parameter email dan password
    public function __construct($email, $password) 
    {
        // membuat data user baru untuk constructor berisi array 2 dimensi email, password, dan role
        $this->_dataUser = [
            (object) [
                'email'     => "soffankelompok46@gmail.com",
                'password'  => "12345",
                'role'      => "superadmin"
            ],
            (object) [
                'email'     => "firmankelompok46@gmail.com",
                'password'  => "12345",
                'role'      => "user"
            ]
        ];
        // membuat email dan password untuk constructor dari parameter yang diberikan
       $this->email = $email;
       $this->password = $password;
    }

    // function login dengan modifer public agar dapat dipanggil dari class lain
    public function login()
    {
        // memanggil function checkCredentials dan disimpan dalam variabel user
        $user = $this->checkCredentials();
        // mengecek $user sesuai dengan data user dan bernilai true
        if($user) {
            // jika bernilai true, memanggil function getRole untuk menyimpan role user
            $this->getRole = $user->role;
            // mengembalikan email user dalam variabel get_data
            return $get_data = $user->email;
        } else {
            // jika bernilai false akan mengembalikan boolean false
            return false;
        }
    }

    // function checkCredentials dengan modifier protected agar tidak dapat digunakan dari class lain
    protected function checkCredentials()
    {
        // perulangan foreach untuk setiap email, password yang ada di data user
        foreach($this->_dataUser as $key => $value) {
            // mengecek apakah value email dan password pada data sama dengan isi parameter dari constructor
            if($value->email == $this->email && $value->password == $this->password) {
                // mengembalikan value berisi email, password, dan role
                return $value;
            }
        }
        // mengembalikan boolean false
        return false;
    }

    // function getRole dengan modifier public
    public function getRole()
    {
        // saat function dipanggil akan mengembalikan nilai role
        return $this->getRole;
    }
}
?>

File: Python/userService.py
# class dengan nama userService
class userService(object):
    # constructor baru __init__ dengan parameter self, email, dan password
    def __init__(self, email, password):
        # menset current instance email dan password dengan paramater
        self.email = email
        self.password = password
        # membuat data baru bernilai array 2 dimensi berisi email, password, dan role user yang bisa login
        self.data = {
            "soffankelompok46@gmail.com" : {
                "email"     : "soffankelompok46@gmail.com",
                "password"  : "12345",
                "role"      : "superadmin"
            },
            "firmankelompok46@gmail.com" : {
                "email"     : "firmankelompok46@gmail.com",
                "password"  : "12345",
                "role"      : "user"
            }
        }
    # function login dengan parameter self
    def login(self):
        # memanggil function checkCredentials dengan variable get_data
        get_data = self.checkCredentials()
        # mengecek apakah function checkCredentials mengembalikan nilai
        if get_data:
            # jika true, maka akan menampilkan welcome role, dan email user yang login
            print("\nWelcome ",get_data['role'])
            print("Logged it as user email: ",get_data['email'])
        else:
            # jika false maka akan menampilkan invalid login
            print("\nInvalid Login!\n")
    #function checkCredentials dengan parameter self
    def checkCredentials(self):
        # membuat perulangan untuk setiap nilai didalam data
        for value in self.data:
            # jika nilai dalam data sama dengan isi paramater email
            if value == self.email:
                # maka akan menyimpan info user yang login kedalam get_data_user
                get_data_user = self.data[value]
                # jika isi parameter password sama dengan password dalam data user
                if self.password == get_data_user['password']:
                    # maka akan mengembalikan data user
                    return get_data_user
                else:
                    #jika tidak maka akan mengembalikkan boolean false
                    return False